aoc/src/main/scala/aoc2022/exercise09.scala

```diff
@@ -0,0 +1,80 @@
+package aoc2022
+import aoc2022.standard.readInput
+import scala.io.Source
+import scala.collection.mutable.Map
+
+
+
+
+object exercise09 extends App:
+
+  case class Coord(hc: Int, vc: Int)
+  case class Grid(h: Coord,allTails: List[Coord], visited: List[Coord] = List.empty):
+    def actionH(action: String): Grid =
+        // In object or in case class?
+        action match
+          case s"${d} ${t}" => (d*t.toInt).split("")
+                              .toList
+                              .foldLeft(this)((g, a) => Grid.followHead(g.copy(h=Grid.moveGrid(g.h,a))))
+          // Is it possible to make the line above cleaner??
+          case _ => sys.error("wrong input ")
+
+  object Grid:
+    def moveGrid(c: Coord, direction: String): Coord =
+      // In object or in case class?
+
+      direction match
+        case "L" => Coord(c.hc - 1, c.vc)
+        case "R" => Coord(c.hc + 1, c.vc)
+        case "U" => Coord(c.hc, c.vc +1)
+        case "D" => Coord(c.hc , c.vc - 1)
+
+    def followHead(g: Grid): Grid =
+      // In object or in case class?
```

Comment:
acts on `g` - thus can be coded as an instance method `followHead: Grid` on the case class